datafusion/proto/src/logical_plan/to_proto.rs

```diff
@@ -1189,24 +1188,79 @@ impl TryFrom<&ScalarValue> for protobuf::ScalarValue {
                     schema: Some(schema),
                 };
 
-                match val {
-                    ScalarValue::List(_) => Ok(protobuf::ScalarValue {
-                        value: Some(protobuf::scalar_value::Value::ListValue(
-                            scalar_list_value,
-                        )),
-                    }),
-                    ScalarValue::LargeList(_) => Ok(protobuf::ScalarValue {
-                        value: Some(protobuf::scalar_value::Value::LargeListValue(
-                            scalar_list_value,
-                        )),
-                    }),
-                    ScalarValue::FixedSizeList(_) => Ok(protobuf::ScalarValue {
-                        value: Some(protobuf::scalar_value::Value::FixedSizeListValue(
-                            scalar_list_value,
-                        )),
-                    }),
-                    _ => unreachable!(),
-                }
+                Ok(protobuf::ScalarValue {
+                    value: Some(protobuf::scalar_value::Value::ListValue(
+                        scalar_list_value,
+                    )),
+                })
+            }
+            ScalarValue::LargeList(arr) => {
```

Comment:
Would it be possible to avoid the copy/paste in this method by making a function? Something like the following?

```rust
fn encode_scalar_list_value(array: &Array) -> Result<protobuf::ScalarListValue> {
...
}
```

Reply:
This one was easy enough to implement.